### ISSUE TITLE
python39Packages.jupyterlab fails to build under channel nixos 22.05

### ISSUE THREAD

**Issue by schlichtanders** [NixOS/nixpkgs#181748]
### Describe the bug
Please help, I don't understand that a package won't build under the current stable nixos channel

### Steps To Reproduce
1. try to install `python39Packages.jupyterlab`
2. it will fail with some error described here https://discourse.nixos.org/t/nixos-22-05-upgrade-fails-with-jupyter-server-and-permissionerror-errno-13-permission-denied-homeless-shelter/20376

### Expected behavior
Every package on a stable nixos channel should have been tested to build under nixos.

### Screenshots
please see the issue
https://discourse.nixos.org/t/nixos-22-05-upgrade-fails-with-jupyter-server-and-permissionerror-errno-13-permission-denied-homeless-shelter/20376

### Additional context
Can someone explain why this was not tested to build properly?

### Notify maintainers
@zimbatm @costrouc 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
/nix/store/9m5k22zdy4s1yh003ma0igqmmrvncww1-bash-interactive-5.1-p16-dev
copying path '/nix/store/9m5k22zdy4s1yh003ma0igqmmrvncww1-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
- system: `"x86_64-linux"`
- host os: `Linux 5.17.5, NixOS, 22.05 (Quokka), 22.05.1711.c06d5fa9c60`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.8.1`
- channels(root): `"nixos-22.05, nixos-unstable"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by bjornfor** [comment]
I bisected a bit on master. It appears that b9f50b7803a0d85bea6ef51ab3a75ec43add7a57 broke it (`nix-build -A python3.pkgs.jupyterlab`), but then sometime after that it got fixed (TODO: bisect that). IOW, maybe we can backport the fix.

**Comment by bjornfor** [comment]
> Please help, I don't understand that a package won't build under the current stable nixos channel

Only a subset of nixpkgs is required to build for the channel to move forward. Apparently `python3Packages.jupyterlab` is not in that subset of [release critical packages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/release.nix#L90).

Also note that nixpkgs and nixos has different criterias for moving their respective channels forward (AFAIK, `pkgs/top-level/release.nix` and `nixos/release*.nix`), and that only _NixOS_ has a "22.05" branch/channel -- nixpkgs has only the unstable channel.

**Comment by bjornfor** [comment]
jupyterlab was fixed on the master branch by cd414d016b2711f36627823f76cbef7e6ffccf85. There's now a backport PR for the fix on branch release-22.05: https://github.com/NixOS/nixpkgs/pull/181769.

**Comment by bjornfor** [comment]
Closing this as https://github.com/NixOS/nixpkgs/pull/181769 was merged.

You can track when the fix arrives in nixos-22.05 here: https://nixpk.gs/pr-tracker.html?pr=181769.

**Comment by schlichtanders** [comment]
very impressive. Thank you @bjornfor for your instant help. Amazing.
